### ISSUE TITLE
BUG On Refreshing FeedView- MVUX (Desktop Platform)

### ISSUE THREAD

**Issue by modula2019** [unoplatform/uno.extensions#2444]
## Current behavior
When a button is clicked to refresh, no refreshing happens. The button is disabled and an error is generated.

Error

```
Exception thrown: 'System.ObjectDisposedException' in Uno.Extensions.Reactive.dll
Uno.Extensions.Reactive.UI.FeedView.RefreshCommand: Warning: Failed to send a refresh request

System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'RequestSource'.
   at Uno.Extensions.Reactive.Core.RequestSource.Send(IContextRequest request)
   at Uno.Extensions.Reactive.Core.RequestSourceExtensions.RequestRefresh(IRequestSource requests)
   at Uno.Extensions.Reactive.UI.FeedView.Subscription.RequestRefresh(Action completionAction)
   at Uno.Extensions.Reactive.UI.FeedView.RefreshCommand.<>c__DisplayClass17_0.<Execute>b__0()
Exception thrown: 'System.InvalidOperationException' in Uno.UI.dll
Exception thrown: 'System.InvalidOperationException' in System.Private.CoreLib.dll      
```  

## Expected behavior

This works well in WinAppSdk. The expected behavior should refresh and display new data from the IState.

## How to reproduce it (as minimally and precisely as possible)
Sample app will be attached. In the app, click on the People Page item on the menu. Then, click on the refresh red icon at the top right side of the opened page.

## Environment

Nuget Package (s):

Package Version(s):

Affected platform(s):

Tested only on Desktop platform

- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Android
- [ ] WebAssembly
- [ ] Windows
- [ ] Skia (WPF)
- [X] Skia (GTK on Linux/macOS/Windows)
- [ ] Skia (Linux Framebuffer)
- [ ] Build tasks

Visual Studio:

- [ ] 2019 (version: )
- [X] 2022 (version: )
- [ ] Visual Studio Code (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )

Relevant plugins:

- [ ] Resharper (version: )

## Anything else we need to know?

None

[UnoApp1.zip](https://github.com/user-attachments/files/16275793/UnoApp1.zip)

**Comment by dr1rrb** [comment]
Hey, I'm unable to get the exact same error that you have, but I got some exceptions with the same effect (button remains disabled after one -or more- clicks) https://github.com/unoplatform/uno.extensions/pull/2448

**Comment by modula2019** [comment]
@dr1rrb did you use the same repro attached? please kindly share the error you got. Thanks

**Comment by modula2019** [comment]
@dr1rrb Yes, that is the error. Do you have a solution for it at the moment? Thanks

**Comment by dr1rrb** [comment]
Unfortunately no workaround for this ... the fix should be merged somehow today and will be available in the dev packages on nuget right after

**Comment by modula2019** [comment]
Ok thanks

**Comment by dr1rrb** [comment]
@modula2019 the fix will also be backported on 4.2 so no needs to switch to dev packages.

**Comment by modula2019** [comment]
@dr1rrb alright. Thanks

**Comment by modula2019** [comment]
@dr1rrb Please when is the stable 4.2 going to be available? I thought it had already been deployed but just realised it is not. 
If you are not doing it anytime soon, what dev package can I use instead?
Thanks

**Comment by modula2019** [comment]
Hi Danield,
Please when is the stable 4.2 going to be available? I thought it had already been deployed but just realised it is not.
If you are not doing it anytime soon, what dev package can I use instead?
Thanks


Best Regards,




[cid:5cadf078-9869-4e73-8928-0eae92938fbd]

Customers Order Management System<https://www.modula-consulting.co.za/products-corms/>

________________________________

**Comment by jhanvi03** [comment]
@dr1rrb, The issue still persist with Uno.sdk & template 5.3.112.
OS: Windows
IDE: VS
Target Solution: Desktop

System.InvalidOperationException: The dependency property system should not be accessed from non UI thread.
   at Microsoft.UI.Xaml.DependencyProperty.GetProperty(Type type, String name) in /__w/1/s/src/Uno.UI/UI/Xaml/DependencyProperty.cs:line 356
   at Microsoft.UI.Xaml.DependencyObjectStore.CallChildCallback(DependencyObjectStore childStore, ManagedWeakReference instanceRef, DependencyProperty property, DependencyPropertyChangedEventArgs eventArgs) in /__w/1/s/src/Uno.UI/UI/Xaml/DependencyObjectStore.cs:line 1988
   at Microsoft.UI.Xaml.DependencyObjectStore.InvokeCallbacks(DependencyObject actualInstanceAlias, DependencyProperty property, DependencyPropertyDetails propertyDetails, Object previousValue, DependencyPropertyValuePrecedences previousPrecedence, Object newValue, DependencyPropertyValuePrecedences newPrecedence, Boolean bypassesPropagation) in /__w/1/s/src/Uno.UI/UI/Xaml/DependencyObjectStore.cs:line 1930
   at Microsoft.UI.Xaml.DependencyObjectStore.RaiseCallbacks(DependencyObject actualInstanceAlias, DependencyPropertyDetails propertyDetails, Object previousValue, DependencyPropertyValuePrecedences previousPrecedence, Object newValue, DependencyPropertyValuePrecedences newPrecedence) in /__w/1/s/src/Uno.UI/UI/Xaml/DependencyObjectStore.cs:line 1857
   at Microsoft.UI.Xaml.DependencyObjectStore.InnerSetValue(DependencyProperty property, Object value, DependencyPropertyValuePrecedences precedence, DependencyPropertyDetails propertyDetails, Boolean isPersistentResourceBinding) in /__w/1/s/src/Uno.UI/UI/Xaml/DependencyObjectStore.cs:line 528
   at Microsoft.UI.Xaml.DependencyObjectStore.CoerceValue(DependencyProperty property) in /__w/1/s/src/Uno.UI/UI/Xaml/DependencyObjectStore.cs:line 696
   at Microsoft.UI.Xaml.Controls.Control.SuppressIsEnabled(Boolean suppress) in /__w/1/s/src/Uno.UI/UI/Xaml/Controls/Control/Control.cs:line 163
   at Microsoft.UI.Xaml.Controls.Primitives.ButtonBase.UpdateCanExecute() in /__w/1/s/src/Uno.UI/UI/Xaml/Controls/Primitives/ButtonBase/ButtonBase.mux.cs:line 281
   at Microsoft.UI.Xaml.Controls.Primitives.ButtonBase.<OnCommandChanged>g__CanExecuteChangedHandler|68_0(Object sender, Object args) in /__w/1/s/src/Uno.UI/UI/Xaml/Controls/Primitives/ButtonBase/ButtonBase.mux.cs:line 236
   at Uno.Extensions.Reactive.UI.FeedView.RefreshCommand.set_IsExecuting(Boolean value)
   at Uno.Extensions.Reactive.UI.FeedView.RefreshCommand.<>c__DisplayClass17_0.<Execute>b__1()
   at Uno.Extensions.Reactive.Core.TokenSetAwaiter`1.Awaiter.Complete()
   at Uno.Extensions.Reactive.Core.TokenSetAwaiter`1.Awaiter.Enable(TokenSetAwaiter`1 owner, TokenSet`1 tokens)
   at Uno.Extensions.Reactive.Core.TokenSetAwaiter`1.Awaiter..ctor(TokenSetAwaiter`1 owner, TokenSet`1 tokens, Action completion)
   at Uno.Extensions.Reactive.Core.TokenSetAwaiter`1.WaitFor(TokenSet`1 tokens, Action completionAction)
   at Uno.Extensions.Reactive.UI.FeedView.Subscription.RequestRefresh(Action completionAction)
   at Uno.Extensions.Reactive.UI.FeedView.RefreshCommand.<>c__DisplayClass17_0.<Execute>b__0()

**Comment by agneszitte** [comment]
> @dr1rrb, The issue still persist with Uno.sdk & template 5.3.112. OS: Windows IDE: VS Target Solution: Desktop

@jhanvi03, after taking a closer look at the sample app attached to this issue, I noticed it is using Uno.Sdk 5.2.175 with additional version overrides, such as for extensions, specified in the Directory.Build.props file.

When upgrading from an Uno Platform 5.2 project, those overrides in Directory.Build.props can be removed, as [explained in the documentation](https://platform.uno/docs/articles/migrating-from-previous-releases.html#single-project-updates).

It seems the override wasn’t noticed when testing the sample app with the upcoming 5.3.112 version.

I ran a test on my side after removing the overrides in Directory.Build.props and was able to reproduce the exception with 5.3.108. However, the exception did not occur with the upcoming Uno.Sdk 5.3.112, that includes the fix from the forthcoming Uno.Extensions 4.2.14 version.

Based on this, I am closing the issue.
